packages/editor/src/components/post-taxonomies/hierarchical-term-selector.js

```diff
@@ -325,18 +264,27 @@ class HierarchicalTermSelector extends Component {
 export default compose( [
 	withSelect( ( select, { slug } ) => {
 		const { getCurrentPost } = select( 'core/editor' );
-		const { getTaxonomy } = select( 'core' );
+		const { getTaxonomy, getEntityRecords } = select( 'core' );
+		const { isResolving } = select( 'core/data' );
 		const taxonomy = getTaxonomy( slug );
+		const availableTerms = getEntityRecords( 'taxonomy', slug, DEFAULT_QUERY );
+		const availableTermsTree = buildTermsTree( availableTerms );
```

Comment:
We will be generating a new reference inside builTermsTree which will probably trigger lost of unnecessary rerenders. As possible solutions, we may create a memoized selector that returns the tree already build. Or create a local memoized one instance of build terms tree, I thinl something like may work:
```
export default compose( [
    (() => {
        const memoizedBuildtermsTree = memize( buildTermsTree, { maxSize: 1 } );
        return withSelect( ( select, { slug } ) => {
            ....
        } );
    })()
```